### PR TITLE
main/llvm9: new aport

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

File: community/lld/APKBUILD [hunk 1]
```diff
@@ -18,7 +18,7 @@ makedepends="
 	zlib-dev"
 checkdepends="gtest gtest-dev bash"
 subpackages="$pkgname-static $pkgname-dev"
-source="https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/lld-$pkgver.src.tar.xz"
+source="http://releases.llvm.org/$pkgver/lld-$pkgver.src.tar.xz"
```
Comment (J0WI) [inline community/lld/APKBUILD]:
```suggestion
source="https://releases.llvm.org/$pkgver/lld-$pkgver.src.tar.xz"
```

File: community/lld/APKBUILD [hunk 2]
```diff
@@ -2,13 +2,13 @@
 # Contributor: Jakub Jirutka <jakub@jirutka.cz>
 # Maintainer: Jakub Jirutka <jakub@jirutka.cz>
 pkgname=lld
-pkgver=8.0.1
+pkgver=9.0.0
 pkgrel=0
 _llvmver=${pkgver%%.*}
 pkgdesc="The LLVM Linker"
 url="https://llvm.org"
```
Comment (J0WI) [inline community/lld/APKBUILD]:
```suggestion
url="https://llvm.org/"
```

File: community/llvm-libunwind/APKBUILD [hunk 3]
```diff
@@ -3,18 +3,18 @@
 # Maintainer: Jakub Jirutka <jakub@jirutka.cz>
 pkgname=llvm-libunwind
 # Note: Update together with llvm.
-pkgver=8.0.1
+pkgver=9.0.0
 _pkgname=libunwind
 _llvmver=${pkgver%%.*}
 pkgrel=0
 pkgdesc="LLVM version of libunwind library"
 url="http://llvm.org/"
```
Comment (J0WI) [inline community/llvm-libunwind/APKBUILD]:
```suggestion
url="https://llvm.org/"
```

File: community/llvm-libunwind/APKBUILD [hunk 4]
```diff
@@ -3,18 +3,18 @@
...
 arch="aarch64 armhf armv7 ppc64le x86 x86_64"
-license="BSD"
+license="Apache-2.0"
 depends_dev="!libunwind-dev"
 makedepends="cmake llvm$_llvmver-dev"
-subpackages="$pkgname-dev"
-source="https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/libunwind-$pkgver.src.tar.xz"
+subpackages="$pkgname-static $pkgname-dev"
+source="http://releases.llvm.org/$pkgver/libunwind-$pkgver.src.tar.xz"
```
Comment (J0WI) [inline community/llvm-libunwind/APKBUILD]:
```suggestion
source="https://releases.llvm.org/$pkgver/libunwind-$pkgver.src.tar.xz"
```

File: main/clang/APKBUILD [hunk 5]
```diff
@@ -3,31 +3,32 @@
 # Maintainer: Natanael Copa <ncopa@alpinelinux.org>
 pkgname=clang
 # Note: Update together with llvm.
-pkgver=8.0.1
+pkgver=9.0.0
 pkgrel=0
 _llvmver=${pkgver%%.*}
 pkgdesc="A C language family front-end for LLVM"
 arch="all"
 url="http://llvm.org/"
```
Comment (J0WI) [inline main/clang/APKBUILD]:
```suggestion
url="https://llvm.org/"
```

File: main/clang/APKBUILD [hunk 6]
```diff
@@ -3,31 +3,32 @@
...
+	llvm$_llvmver-test-utils
 	ninja
 	python3
 	"
 checkdepends="diffutils bash"
 depends_dev="$pkgname=$pkgver-r$pkgrel"
 subpackages="$pkgname-static $pkgname-dev $pkgname-doc $pkgname-libs
 	$pkgname-analyzer::noarch"
-source="https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver/cfe-$pkgver.src.tar.xz
+source="http://releases.llvm.org/$pkgver/cfe-$pkgver.src.tar.xz
```
Comment (J0WI) [inline main/clang/APKBUILD]:
```suggestion
source="https://releases.llvm.org/$pkgver/cfe-$pkgver.src.tar.xz
```

File: main/compiler-rt/APKBUILD [hunk 7]
```diff
@@ -21,8 +21,8 @@ makedepends="
 checkdepends="gtest-dev llvm-test-utils>=$_llvmver"
 subpackages="$pkgname-static"
 options="!check" # Doesn't seem to work with linux ld: /lib/ld-musl-x86_64.so.1: cannot load --version: No such file or directory
-source="https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver/compiler-rt-$pkgver.src.tar.xz
-	https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver/llvm-$pkgver.src.tar.xz"
+source="http://releases.llvm.org/$pkgver/compiler-rt-$pkgver.src.tar.xz
```
Comment (J0WI) [inline main/compiler-rt/APKBUILD]:
```suggestion
source="https://releases.llvm.org/$pkgver/compiler-rt-$pkgver.src.tar.xz
```

File: main/compiler-rt/APKBUILD [hunk 8]
```diff
@@ -21,8 +21,8 @@ makedepends="
...
+	http://releases.llvm.org/$pkgver/llvm-$pkgver.src.tar.xz"
```
Comment (J0WI) [inline main/compiler-rt/APKBUILD]:
```suggestion
	https://releases.llvm.org/$pkgver/llvm-$pkgver.src.tar.xz"
```

File: main/llvm9/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+pkgdesc="Low Level Virtual Machine compiler system, version $_majorver"
+arch="all"
+url="https://llvm.org/"
+license="Apache-2.0"
+depends_dev="$pkgname=$pkgver-r$pkgrel"
+makedepends="binutils-dev chrpath cmake libffi-dev paxmark python3 zlib-dev
+	libxml2-dev"
+checkdepends="bash py3-setuptools coreutils" # coreutils for 'od' binary
+subpackages="$pkgname-static $pkgname-libs $pkgname-dev $pkgname-test-utils:_test_utils"
+source="http://releases.llvm.org/$pkgver/llvm-$pkgver.src.tar.xz
```
Comment (J0WI) [inline main/llvm9/APKBUILD]:
```suggestion
source="https://releases.llvm.org/$pkgver/llvm-$pkgver.src.tar.xz
```

File: testing/lldb/APKBUILD [hunk 10]
```diff
@@ -25,10 +25,9 @@ makedepends="
 	python2-dev
 	swig
 	"
-subpackages="$pkgname-static $pkgname-dev py2-$pkgname:py2"
-source="https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/lldb-8.0.1.src.tar.xz
+subpackages="$pkgname-dev py2-$pkgname:py2"
+source="http://releases.llvm.org/$pkgver/lldb-$pkgver.src.tar.xz
```
Comment (J0WI) [inline testing/lldb/APKBUILD]:
```suggestion
source="https://releases.llvm.org/$pkgver/lldb-$pkgver.src.tar.xz
```

File: community/lld/APKBUILD (hunk 2 above)
Comment (Cogitri) [reply community/lld/APKBUILD]: Does the trailing slash really make a difference?

**Comment by Cogitri** [comment]
Thanks for taking a look, @J0WI !

**Comment by J0WI** [comment]
`firefox-esr` can be recompiled against LLVM 9

**Comment by Cogitri** [comment]
Yes, so can firefox and thunderbird, but I want to do that in a seperate PR once this is good to go.

File: community/lld/APKBUILD (hunk 2 above)
Comment (J0WI) [reply community/lld/APKBUILD]: Not really, it's just to normalize

**Comment by Cogitri** [comment]
Merged by now.

**Pull request closed without merging**